### PR TITLE
feat: update CLI for gpu and statistics tools

### PR DESCRIPTION
**Pull request by muchvo**
## Description

1. Update CLI for gpu and statistics tools.

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [ ] I have raised an issue to propose this change ([required](https://github.com/PKU-MARL/omnisafe/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://omnisafe.readthedocs.io/en/latest/developer/contributing.html) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.
- [ ] I have reformatted the code using `make format`. (**required**)
- [ ] I have checked the code using `make lint`. (**required**)
- [ ] I have ensured `make test` pass. (**required**)

### CODE REVIEW

File: omnisafe/utils/command_app.py [hunk 1]
```diff
@@ -200,11 +240,37 @@ def benchmark(
...
+    if gpu_range is not None:
+        assert_with_exit(
+            len(gpu_range.split(':')) <= 3,
+            'gpu_range must be like 0:2, which means using gpu 0 and 1',
+        )
+        # Set the device.
+        avaliable_gpus = list(range(torch.cuda.device_count()))
+        gpu_id = list(range(*[int(i) for i in gpu_range.split(':')]))
+
+        if set(gpu_id) > set(avaliable_gpus):
```
Comment (XuehaiPan) [inline omnisafe/utils/command_app.py:254]: Wrong condition here.

**Review by zmsn-2077 — APPROVED** [review]
LGTM.

**Review by Gaiejj — APPROVED** [review]
LGTM.

File: examples/benchmarks/run_experiment_grid.py [hunk 2]
```diff
@@ -93,7 +93,7 @@ def train(
     # if you want to use CPU, please set gpu_id = None
     # gpu_id = None
 
-    if set(gpu_id) > set(avaliable_gpus):
+    if not set(gpu_id).issubset(set(avaliable_gpus)):
```
Comment (XuehaiPan) [inline examples/benchmarks/run_experiment_grid.py:96]:
```suggestion
    if not set(gpu_id).issubset(avaliable_gpus):
```

File: omnisafe/utils/command_app.py [hunk 3]
```diff
@@ -200,11 +240,38 @@ def benchmark(
...
+        assert_with_exit(
+            len(gpu_range.split(':')) <= 3,
+            'gpu_range must be like x:y:z format,'
+            ' which means using gpu in [x, y) with step size z',
+        )
+        # Set the device.
+        avaliable_gpus = list(range(torch.cuda.device_count()))
+        gpu_id = list(range(*[int(i) for i in gpu_range.split(':')]))
+
+        if not set(gpu_id).issubset(set(avaliable_gpus)):
```
Comment (XuehaiPan) [inline omnisafe/utils/command_app.py:255]:
```suggestion
        if not set(gpu_id).issubset(avaliable_gpus):
```

**Pull request merged**